src/lib/y2security/lsm/config.rb

```diff
@@ -0,0 +1,167 @@
+# Copyright (c) [2021] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "yast2/execute"
+require "y2security/lsm/base"
+require "y2security/lsm/app_armor"
+require "y2security/lsm/none"
+require "y2security/lsm/selinux"
+
+Yast.import "Stage"
+Yast.import "ProductFeatures"
+
+module Y2Security
+  module LSM
+    # This class allows to check and select one of the supported Linux Security Modules (LSM)
+    class Config
+      include Yast::Logger
+      include Singleton
+
+      RUNNING_PATH = "/sys/kernel/security/lsm".freeze
+      SUPPORTED = [None, Selinux, AppArmor].freeze
+
+      # @return [None, AppArmor,Selinux, nil] selected module
+      attr_accessor :selected
+      # @return [Boolean] Whether LSM can be configured by the user or not
+      attr_accessor :configurable
+
+      def initialize
+        supported.each { |m| self.class.send(:define_method, m.id) { m } }
+      end
+
+      # Select the LSM to be used based in the one defined in the control file using apparmor as
+      # fallback in case that no one is selected
+      def propose_default
+        log.info("The settings are #{product_feature_settings.inspect}")
+        selected = product_feature_settings.fetch(:select, "apparmor")
+
+        select(selected)
+      end
+
+      # Select the module with the given id when it is selectable.
+      #
+      # @param id [String,Symbol] LSM id to be selected
+      def select(id)
+        @selected = selectable.find { |m| m.id == id.to_sym }
+        log.info @selected ? "Selected LSM with id: #{id}" : "No LSM selected with id: #{id}"
+        @selected
+      end
+
+      # Convenience method to obtain a list of the supported and selectable modules
+      #
+      # @return [Array<Y2Security::LSM::Base] array of supported and selectable LSMs
+      def selectable
+        supported.select(&:selectable?)
+      end
+
+      # Returns the needed patterns for the selected LSM or an empty array if no one is selected
+      #
+      # @return [Array<Sting>]
+      def needed_patterns
+        return [] unless selected
+
+        selected.needed_patterns
+      end
+
+      # Convenience method to save the configuration for the selected LSM
```

Comment:
```suggestion
      # Convenience method to save the configuration for the selected LSM
      #
      # @note: all LSM kernel options are reset before save the selected one. See {Y2Security::LSM::Base#reset_kernel_params}
      #
      # @return [Boolean] true if config is successfully save; false otherwise.
```